src/common/types.rs

```diff
@@ -227,30 +227,30 @@ impl Ty
   }
 }
 
-pub fn u(size: u32) -> &'static Ty
+pub fn u(size: usize) -> &'static Ty
 {
   match size {
-    8  => U8,
+    1  => U8,
     _  => panic!("unsupported integer type: i{}", size)
   }
 }
 
-pub fn i(size: u32) -> &'static Ty
+pub fn i(size: usize) -> &'static Ty
```

Comment:
Ok. If so, how about changing the ```size_of()``` function to return the number of bits in ```u32``` type?